Simplify list-group borders in cards

mdo opened the pull request:
Consolidates some CSS and beefs up the selector to account for inherited border styles. Fixes #30793, a regression from v4.4.1 that shipped in #30497.

XhmikosR commented:
Adding the comment from the issue so that we triple check before merging https://github.com/twbs/bootstrap/issues/30793#issuecomment-628354971

mdo committed:
Simplify list-group borders to prevent double borders in cards

Consolidates some CSS and beefs up the selector to account for inherited border styles. Fixes #30793, a regression from v4.4.1 that shipped in #30497.

MartijnCuppens reviewed (approved):
Proper solution 👌

MartijnCuppens committed:
Merge branch 'master' into card-list-group-borders
